providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -447,6 +457,27 @@ public boolean apply(OperatingSystem input) {
               .first();
    }
 
+   private String getNotesFrom(SoftLayerTemplateOptions templateOptions) {
+      if (templateOptions.getNotes().isPresent()) {
+         return templateOptions.getNotes().get();
+      } else {
+         Map<String, String> meta = templateOptions.getUserMetadata();
+         if (meta != null && !meta.isEmpty()) {
+            if (meta.containsKey(USER_META_NOTES)) {
+               return meta.get(USER_META_NOTES);
+            } else {
+               String notes = "User Metadata\n===========\n\n" + Joiner.on("\n").withKeyValueSeparator(": ").join(meta);
+               if (notes.length() > SoftLayerTemplateOptions.NOTES_MAX_LENGTH) {
+                  String suffix = "...\n<truncated>";
+                  notes = notes.substring(0, SoftLayerTemplateOptions.NOTES_MAX_LENGTH - suffix.length()) + suffix;
+               }
+               return notes;
+            }
+         }
+      }
+      return null;
+   }
```

Comment:
The canonical way to set the notes is through the property on the `SoftlayerTemplateOptions`. As the user metadata is part of the generic API it might be easier for users to write provider-independent code by setting the notes there, but don't mind removing it if you think it's too much.

Softlayer doesn't support user metadata, so the values there are ignored. The notes looked like a good place to store this information if not overridden by an explicit value. That was my thinking behind the various ways of getting the notes - wdyt?